supervisor/issue_handler.py

```diff
@@ -40,29 +44,111 @@ def resolve_set_status(self, status: IssueStatus, why: str):
+                    f"redhat/{group}/{component}",
+                    issue.key,
+                    state=MergeRequestState.MERGED,
+                )
 
-        logger.info("Running workflow for issue %s", issue.url)
+                if merged_mr := next(merged_mrs, None):
+                    add_issue_label(
+                        issue.key,
+                        JotnarLabel.MERGED,
+                        f"A [merge request| {merged_mr.url}]. resolving this issue has been merged; waiting for errata creation and final testing.",
+                        dry_run=self.dry_run,
+                    )
+
+                    issue.labels.append(JotnarLabel.MERGED)
+                    return True
+
+        return False
+
+    def get_latest_merged_timestamp(self):
+        """This function will return DATETIME_MIN_UTC if it doesn't find any merged MRs"""
+        issue = self.issue
+        component = issue.components[0]
+
+        def get_merged_mrs_for_group():
+            for group in GITLAB_GROUPS:
+                project = f"redhat/{group}/{component}"
+                for mr in search_gitlab_project_mrs(
+                    project,
+                    issue.key,
+                    state=MergeRequestState.MERGED,
+                ):
+                    yield mr
+
+        return max(
+            (mr.merged_at or DATETIME_MIN_UTC for mr in get_merged_mrs_for_group()),
+            default=DATETIME_MIN_UTC,
+        )
 
-        if "jotnar_needs_attention" in issue.labels:
+    async def run_before_errata_created(self) -> WorkflowResult:
+        issue = self.issue
+
+        # All issues in this branch should have no errata link
+
+        if not any(
+            label in (JotnarLabel.BACKPORTED, JotnarLabel.REBASED, JotnarLabel.MERGED)
+            for label in issue.labels
+        ):
             return self.resolve_remove_work_item(
-                "Issue has the jotnar_needs_attention label"
+                f"Issue without target labels: {issue.labels}"
             )
 
-        if issue.errata_link is None:
-            return self.resolve_remove_work_item("Issue has no errata_link")
+        if JotnarLabel.MERGED not in issue.labels:
+            self.label_merge_if_needed()
 
+        if JotnarLabel.MERGED not in issue.labels:
+            return self.resolve_wait(
+                "No merged MR found, reschedule it for 3 hours",
+                reschedule_in=60 * 60 * 3,
+            )
+
+        latest_merged_timestamp = self.get_latest_merged_timestamp()
+        cur_time = datetime.now(tz=timezone.utc)
+        time_diff = abs(cur_time - latest_merged_timestamp)
+        if time_diff < timedelta(days=1):
+            return self.resolve_wait(
+                "Wait for the associated erratum to be created",
+                reschedule_in=60 * 60,
+            )
+        else:
+            return self.resolve_flag_attention(
+                "No errata link found after 24 hours after the MR got merged"
```

Comment:
this needs to be a bit more verbose - it's appearing in a public comment.

 A merge request was merged for this issue more than 24 hours ago but no errata
 was created. Please investigate and look for gating failures or other reasons that
 might have blocked errata creation.